packages/app/src/file-system/services/file-system.service.refactored.ts

```diff
@@ -0,0 +1,141 @@
+  }
+
+  public async extractFile(
+    sourceFile: string,
+    destinationDirectory: string
+  ): Promise<string[]> {
+    const input = new stream.PassThrough()
+    const file = this.readFile(sourceFile)
+    input.end(file)
+
+    const entryFilePaths: string[] = []
+    const extract = tar.extract({ allowUnknownFormat: true })
+
+    extract.on("entry", function (header, inputStream, next) {
+      const entryFilePath = path.join(destinationDirectory, header.name)
+      const output = fs.createWriteStream(entryFilePath)
+      inputStream.pipe(output)
+      inputStream.on("end", function () {
+        next()
+      })
+      inputStream.resume()
+      entryFilePaths.push(entryFilePath)
+    })
+
+    input.pipe(extract)
+
+    return this.handleStreamFinish(input).then(() => entryFilePaths)
+  }
+
+  public async extractEncryptedFile(
+    sourceFile: string,
+    destinationDirectory: string,
+    token: string
+  ): Promise<string[]> {
+    const input = new stream.PassThrough()
+    const file = this.readFile(sourceFile)
+    input.end(file)
+
+    const entryFilePaths: string[] = []
+    const extract = tar.extract({ allowUnknownFormat: true })
+
+    extract.on("entry", function (header, inputStream, next) {
+      const entryFilePath = path.join(destinationDirectory, header.name)
+      const iv = crypto.randomBytes(16)
+      const cipher = CryptoFileService.createCipherivViaToken({ token, iv })
+      const appendIv = new AppendIv(iv)
+      const output = fs.createWriteStream(entryFilePath)
+      inputStream.pipe(cipher).pipe(appendIv).pipe(output)
+      inputStream.pipe(output)
+      inputStream.on("end", function () {
+        next()
+      })
+      inputStream.resume()
+      entryFilePaths.push(entryFilePath)
+    })
+
+    input.pipe(extract)
+
+    return this.handleStreamFinish(input).then(() => entryFilePaths)
+  }
+
+  public async readEncryptedFile(
+    filePath: string,
+    token: string
+  ): Promise<Buffer | undefined> {
+    const buffer = await this.readFile(filePath)
+    return CryptoFileService.decryptViaToken({ buffer, token })
+  }
+
+  public async writeEncryptedFile(
+    filePath: string,
+    data: Buffer | Uint8Array,
+    token: string
+  ): Promise<void> {
+    const encryptedBuffer = CryptoFileService.encryptViaToken({
+      buffer: data,
+      token,
+    })
+
+    if (encryptedBuffer === undefined) {
+      throw new Error("Encryption fails")
+    }
+
+    const fileDirectory = path.dirname(filePath)
+    const directoryExists = await this.exists(fileDirectory)
+
+    if (!directoryExists) {
+      await this.createDirectory(fileDirectory)
+    }
+
+    return this.writeFile(filePath, encryptedBuffer)
+  }
+
+  private handleStreamFinish(input: Transform): Promise<void> {
+    return new Promise((resolve, reject) => {
+      input
+        .on("finish", function () {
+          resolve()
+        })
+        .on("error", function () {
+          reject()
+        })
+    })
+  }
+}
```

Comment:
Why with `refactored` suffix? I suggest removing suffix and deleting below files:
- sync-file-system.service.ts 
- sync-file-system-class.ts
- sync-file-system-service-factory.ts

Reply:
Can't be implemented because the legacy `FileSystemService` is used across the application

Reply:
where? 🤔

Reply:
https://github.com/mudita/mudita-center/blob/master/packages%2Fapp%2Fsrc%2Fbackend%2Fadapters%2Fdevice-file-system%2Fdevice-file-system.adapter.ts#L265